src/poetry/console/commands/build.py

```diff
@@ -31,6 +32,6 @@ def handle(self) -> int:
             )
 
             builder = Builder(self.poetry)
-            builder.build(fmt, executable=env.python)
+            builder.build(fmt, executable=env.python, target_dir=self.option("output"))
```

Comment:
I haven't tested it, but I'm afraid that now the default behavior would be to use a `dist` folder relative to the current working directory instead of relative to the `pyproject.toml` file.

Furthermore `target_dir` expects `Path` and not `str` :)